soc/xtensa/esp32s2/power.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/pm/pm.h>
+#include <esp_sleep.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+
+/* Invoke Low Power/System Off specific Tasks */
+__weak void pm_state_set(enum pm_state state, uint8_t substate_id)
+{
+	ARG_UNUSED(substate_id);
+
+	switch (state) {
+	case PM_STATE_SOFT_OFF:
+		/* Forces RTC domain to be always on */
+		esp_sleep_pd_config(ESP_PD_DOMAIN_RTC_PERIPH, ESP_PD_OPTION_ON);
+		esp_deep_sleep_start();
+		break;
+	default:
+		LOG_DBG("Unsupported power state %u", state);
+		break;
+	}
+
+	k_cpu_idle();
```

Comment:
why are you calling `k_cpu_idle()` here ? This will be called after the system wakes up and does not look correct, you should just return and let the idle thread runs again. The pm subsystem then will check with the policy manager which power state to use, if there is no suitable state it will return `PM_STATE_ACTIVE` and the idle thread will then call `k_cpu_idle()`.

Reply:
> I have just two concerns, one in `pm_state_set` that is calling `k_cpu_idle` and the other is about not `pm_state_exit_post_ops` not being restoring interruptions. Other than that looks great this pr.

I have removed the `k_cpu_idle` call and included interruption restoring for the PM_STANDBY power state. When SYSTEM_OFF is used, SoC (re)initialization already initializes and restores interruptions.